File: sohee-jung/week1/백준_silver_차집합.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        StringTokenizer st = new StringTokenizer(br.readLine());

        int A = Integer.parseInt(st.nextToken());
        int B = Integer.parseInt(st.nextToken());

        HashSet<Integer> setA = new HashSet<>();
        HashSet<Integer> setB = new HashSet<>();


        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < A; i++) {
            int tempA = Integer.parseInt(st.nextToken());
            setA.add(tempA);
        }

        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < B; i++) {
            int tempB = Integer.parseInt(st.nextToken());
            setB.add(tempB);
        }

        HashSet<Integer> subtract = new HashSet<>(setA);
        subtract.removeAll(setB);

        if (subtract.isEmpty()) {
            bw.write("0\n");
        } else {
            List<Integer> result = new ArrayList<>(subtract);
            Collections.sort(result);

            bw.write(result.size() + "\n");
            for (int i : result) {
                bw.write(i + " ");
            }
        }

        bw.flush();
        bw.close();
    }
}

File: minseo-song/week1/백준_silver_차집합.kt
package week1

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.StringTokenizer

fun main() {
    val bufferedReader = BufferedReader(InputStreamReader(System.`in`))
    // val br = System.in.bufferedReader() 이렇게 하면 import구문 안적어도 된다고 함
    val bufferedWriter = BufferedWriter(OutputStreamWriter(System.out))

    val st = StringTokenizer(bufferedReader.readLine())
    val nA = st.nextToken().toInt()
    val nB = st.nextToken().toInt()

    val setA = mutableSetOf<Int>()
    val stA = StringTokenizer(bufferedReader.readLine())
    repeat(nA) {
        setA.add(stA.nextToken().toInt())
    }

    val setB = mutableSetOf<Int>()
    val stB = StringTokenizer(bufferedReader.readLine())
    repeat(nB) {
        setB.add(stB.nextToken().toInt())
    }

    val result = setA.filter { it !in setB }.sorted()

    bufferedWriter.write(result.size.toString() + "\n")
    if (result.isNotEmpty()){
        repeat(result.size){
            bufferedWriter.write(result[it].toString() + " ")
        }
    }
    bufferedReader.close()
    bufferedWriter.flush() // close만 해도 됨
}